### ISSUE TITLE
Reinstalling wazuh-3.9.2-1 over the ghost of 3.5.0-1 fails

### ISSUE THREAD

**Issue by jsoref** [wazuh/wazuh-api#410]
```
+ '[' -d /var/ossec/packages_files/api_config_files/api/configuration ']'
+ cp /var/ossec/api/configuration/config.js /var/ossec/packages_files/api_config_files/api/configuration/config.js.debnew
+ cp -rfp /var/ossec/packages_files/api_config_files/api/configuration /var/ossec/api/
+ cp -rfp /var/ossec/packages_files/api_config_files/api/configuration/auth/user /var/ossec/api/configuration/auth/user
cp: cannot stat '/var/ossec/packages_files/api_config_files/api/configuration/auth/user': No such file or directory
```

**Comment by druizz90** [comment]
Hi @jsoref,

I installed Wazuh `3.5.0-1` (manager and API) in a Ubuntu system and I upgraded to `3.9.2-1` version without getting errors. 

Please, could you tell me the steps you followed? It was possible that you deleted files in API directory before upgrading Wazuh?

Best regards,

Demetrio.

**Comment by jsoref** [comment]
```
/var/log/apt/history.log.1.gz:

Start-Date: 2019-06-18  21:05:08
Commandline: apt remove python3.6-minimal
Requested-By: jsoref (1001)
Remove: python3-distro-info:amd64 (0.18ubuntu0.18.04.1), python3-json-pointer:amd64 (1.10-1), python3-google-compute-engine:amd64 (20190124+dfsg1-0ubuntu1~18.04.0), netplan.io:amd64 (0.97-0ubuntu1~18.04.1), cloud-init:amd64 (19.1-1- gbaa47854-0ubuntu1~18.04.1), python3-ndg-httpsclient:amd64 (0.4.4-1), update-manager-core:amd64 (1:18.04.11.10), networkd-dispatcher:amd64 (1.7-0ubuntu3.3), python3-zope.interface:amd64 (4.3.2-1+ubuntu18.04.1+certbot+1), python3-dis tutils:amd64 (3.6.8-1~18.04), python3-software-properties:amd64 (0.96.24.32.9), byobu:amd64 (5.125-0ubuntu1), wazuh-api:amd64 (3.9.2-1), update-notifier-common:amd64 (3.192.1.7), python3-distro:amd64 (1.0.1-2), python3-jwt:amd64 (1. 5.3+ds1-1), python3-configargparse:amd64 (0.11.0-1+certbot~xenial+1), python3-systemd:amd64 (234-1build1), python3-josepy:amd64 (1.1.0-2+ubuntu18.04.1+certbot+1), snapd:amd64 (2.38+18.04), pastebinit:amd64 (1.5-2), python3-asn1crypt o:amd64 (0.24.0-1), python3-urllib3:amd64 (1.22-1ubuntu0.18.04.1), certbot:amd64 (0.31.0-1+ubuntu18.04.1+certbot+1), python3-zope.hookable:amd64 (4.0.4-4+ubuntu18.04.1+certbot+1), language-selector-common:amd64 (0.188.2), python3-js onpatch:amd64 (1.19+really1.16-1fakesync1), python3-pbr:amd64 (3.1.1-3ubuntu3), ubuntu-standard:amd64 (1.417.1), python3-netifaces:amd64 (0.10.4-0.1build4), python3.6:amd64 (3.6.7-1~18.04), python3:amd64 (3.6.7-1~18.04), python3-idn a:amd64 (2.6-1), python3-augeas:amd64 (0.5.0-1+ubuntu18.04.1+certbot+1), python3-zope.event:amd64 (4.2.0-1), plymouth-theme-ubuntu-text:amd64 (0.9.3-1ubuntu7.18.04.2), python3-six:amd64 (1.11.0-2), python3-jsonschema:amd64 (2.6.0-2) , zfs-zed:amd64 (0.7.5-1ubuntu16.4), python3-zope.component:amd64 (4.3.0-1+ubuntu18.04.1+certbot+3), python3-mock:amd64 (2.0.0-3), python3-requests:amd64 (2.18.4-2ubuntu0.1), python3-httplib2:amd64 (0.9.2+dfsg-1ubuntu0.1), nplan:amd 64 (0.97-0ubuntu1~18.04.1), python3-pycurl:amd64 (7.43.0.1-0.2), zfsutils-linux:amd64 (0.7.5-1ubuntu16.4), python3.6-minimal:amd64 (3.6.7-1~18.04), ubuntu-minimal:amd64 (1.417.1), apport:amd64 (2.20.9-0ubuntu7.6), python3-future:amd 64 (0.15.2-4+ubuntu18.04.1+certbot+3), python3-minimal:amd64 (3.6.7-1~18.04), python3-cryptography:amd64 (2.1.4-1ubuntu1.2), python3-distupgrade:amd64 (1:18.04.33), python3-debconf:amd64 (1.5.66ubuntu1), python3-commandnotfound:amd6 4 (18.04.5), python3-update-manager:amd64 (1:18.04.11.10), ubuntu-release-upgrader-core:amd64 (1:18.04.33), python3-apport:amd64 (2.20.9-0ubuntu7.6), python3-cffi-backend:amd64 (1.11.5-1), python3-boto:amd64 (2.44.0-1ubuntu2.18.04.0 ), python3-markupsafe:amd64 (1.0-1build1), ufw:amd64 (0.36-0ubuntu0.18.04.1), python3-acme:amd64 (0.31.0-1+ubuntu18.04.1+certbot+1), python3-configobj:amd64 (5.0.6-2+ubuntu18.04.1+certbot+1), python3-blinker:amd64 (1.4+dfsg1-0.1), u buntu-core-launcher:amd64 (2.38+18.04), python3-lib2to3:amd64 (3.6.8-1~18.04), python3-yaml:amd64 (3.12-1build2), python3-pkg-resources:amd64 (39.0.1-2), python3-openssl:amd64 (17.5.0-1ubuntu1), dh-python:amd64 (3.20180325ubuntu2), python3-rfc3339:amd64 (1.0-4+certbot~xenial+1), python3-requests-toolbelt:amd64 (0.8.0-1+ubuntu18.04.1+certbot+1), python3-certbot-apache:amd64 (0.31.0-1+ubuntu18.04.1+certbot+1), sosreport:amd64 (3.6-1ubuntu0.18.04.2), python3-dbus:amd64 (1.2.6-1), python3-prettytable:amd64 (0.7.2-3), lsb-release:amd64 (9.20170808ubuntu1), python3-certbot:amd64 (0.31.0-1+ubuntu18.04.1+certbot+1), python3-gi:amd64 (3.26.1-2ubuntu1), python3-tz:amd64 (2018.3-2), unattended-upgrades:amd64 (1.1ubuntu1.18.04.11), wazuh-manager:amd64 (3.9.2-1), python3-oauthlib:amd64 (2.0.6-1), apparmor:amd64 (2.12-4ubuntu5.1), command-not-found:amd64 (18.04.5), ssh-import-id:amd64 (5.7-0ubuntu1.1), python3-requests-unixsocket:amd64 (0.1.5-3), python3-jinja2:amd64 (2.10-1ubuntu0.18.04.1), python3-debian:amd64 (0.1.32), python3-gdbm:amd64 (3.6.8-1~18.04), python3-newt:amd64 (0.52.20-1ubuntu1), python3-certifi:amd64 (2018.1.18-2), cloud-guest-utils:amd64 (0.30-0ubuntu5), python3-parsedatetime:amd64 (2.4-3+ubuntu18.04.1+certbot+3), python3-chardet:amd64 (3.0.4-1+ubuntu16.04.1+certbot+2), python3-problem-report:amd64 (2.20.9-0ubuntu7.6), gce-compute-image-packages:amd64 (20190124+dfsg1-0ubuntu1~18.04.0), python3-serial:amd64 (3.4-2), python3-apt:amd64 (1.6.4), python3-pyasn1:amd64 (0.4.2-3), software-properties-common:amd64 (0.96.24.32.9)
Error: Sub-process /usr/bin/dpkg returned an error code (1)
End-Date: 2019-06-18  21:05:46

Start-Date: 2019-06-18  21:22:39
Commandline: apt-get install python3-distro-info python3-json-pointer python3-google-compute-engine netplan.io cloud-init python3-ndg-httpsclient update-manager-core networkd-dispatcher python3-zope.interface python3-distutils python3-software-properties byobu wazuh-api update-notifier-common python3-distro python3-jwt python3-configargparse python3-systemd python3-josepy snapd pastebinit python3-asn1crypto python3-urllib3 certbot python3-zope.hookable language-selector-common python3-jsonpatch python3-pbr ubuntu-standard python3-netifaces python3.6 python3 python3-idna python3-augeas python3-zope.event plymouth-theme-ubuntu-text python3-six python3-jsonschema zfs-zed python3-zope.component python3-mock python3-requests python3-httplib2 nplan python3-pycurl zfsutils-linux python3.6-minimal ubuntu-minimal apport python3-future python3-minimal python3-cryptography python3-distupgrade python3-debconf python3-commandnotfound python3-update-manager ubuntu-release-upgrader-core python3-apport python3-cffi-backend python3-boto python3-markupsafe ufw python3-acme python3-configobj python3-blinker ubuntu-core-launcher python3-lib2to3 python3-yaml python3-pkg-resources python3-openssl dh-python python3-rfc3339 python3-requests-toolbelt python3-certbot-apache sosreport python3-dbus python3-prettytable lsb-release python3-certbot python3-gi python3-tz unattended-upgrades wazuh-manager python3-oauthlib apparmor command-not-found ssh-import-id python3-requests-unixsocket python3-jinja2 python3-debian python3-gdbm python3-newt python3-certifi cloud-guest-utils python3-parsedatetime python3-chardet python3-problem-report gce-compute-image-packages python3-serial python3-apt python3-pyasn1 software-properties-common
Install: python3-distro-info:amd64 (0.18ubuntu0.18.04.1), python3-json-pointer:amd64 (1.10-1), python3-google-compute-engine:amd64 (20190124+dfsg1-0ubuntu1~18.04.0), netplan.io:amd64 (0.97-0ubuntu1~18.04.1), cloud-init:amd64 (19.1-1-gbaa47854-0ubuntu1~18.04.1), python3-ndg-httpsclient:amd64 (0.4.4-1), update-manager-core:amd64 (1:18.04.11.10), networkd-dispatcher:amd64 (1.7-0ubuntu3.3), python3-zope.interface:amd64 (4.3.2-1+ubuntu18.04.1+certbot+1), python3-distutils:amd64 (3.6.8-1~18.04), python3-software-properties:amd64 (0.96.24.32.9), python3-icu:amd64 (1.9.8-0ubuntu1, automatic), byobu:amd64 (5.125-0ubuntu1), wazuh-api:amd64 (3.9.2-1), update-notifier-common:amd64 (3.192.1.7), python3-distro:amd64 (1.0.1-2), python3-jwt:amd64 (1.5.3+ds1-1), python3-configargparse:amd64 (0.11.0-1), python3-systemd:amd64 (234-1build1), python3-josepy:amd64 (1.1.0-2+ubuntu18.04.1+certbot+1), snapd:amd64 (2.38+18.04), pastebinit:amd64 (1.5-2), python3-asn1crypto:amd64 (0.24.0-1), python3-urllib3:amd64 (1.22-1ubuntu0.18.04.1), certbot:amd64 (0.31.0-1+ubuntu18.04.1+certbot+1), python3-zope.hookable:amd64 (4.0.4-4+ubuntu18.04.1+certbot+1), language-selector-common:amd64 (0.188.2), python3-jsonpatch:amd64 (1.19+really1.16-1fakesync1), python3-pbr:amd64 (3.1.1-3ubuntu3), ubuntu-standard:amd64 (1.417.1), python3-netifaces:amd64 (0.10.4-0.1build4), python3.6:amd64 (3.6.8-1~18.04.1), python3:amd64 (3.6.7-1~18.04), python3-idna:amd64 (2.6-1), python3-augeas:amd64 (0.5.0-1+ubuntu18.04.1+certbot+1), python3-zope.event:amd64 (4.2.0-1), plymouth-theme-ubuntu-text:amd64 (0.9.3-1ubuntu7.18.04.2), python3-six:amd64 (1.11.0-2), python3-jsonschema:amd64 (2.6.0-2), zfs-zed:amd64 (0.7.5-1ubuntu16.4), python3-zope.component:amd64 (4.3.0-1+ubuntu18.04.1+certbot+3), python3-mock:amd64 (2.0.0-3), python3-requests:amd64 (2.18.4-2ubuntu0.1), python3-httplib2:amd64 (0.9.2+dfsg-1ubuntu0.1), nplan:amd64 (0.97-0ubuntu1~18.04.1), python3-pycurl:amd64 (7.43.0.1-0.2), zfsutils-linux:amd64 (0.7.5-1ubuntu16.4), python3.6-minimal:amd64 (3.6.8-1~18.04.1), ubuntu-minimal:amd64 (1.417.1), apport:amd64 (2.20.9-0ubuntu7.6), python3-future:amd64 (0.15.2-4+ubuntu18.04.1+certbot+3), python3-minimal:amd64 (3.6.7-1~18.04), python3-cryptography:amd64 (2.1.4-1ubuntu1.3), python3-distupgrade:amd64 (1:18.04.33), python3-debconf:amd64 (1.5.66ubuntu1), python3-commandnotfound:amd64 (18.04.5), python3-update-manager:amd64 (1:18.04.11.10), ubuntu-release-upgrader-core:amd64 (1:18.04.33), python3-apport:amd64 (2.20.9-0ubuntu7.6), python3-cffi-backend:amd64 (1.11.5-1), python3-boto:amd64 (2.44.0-1ubuntu2.18.04.0), python3-markupsafe:amd64 (1.0-1build1), ufw:amd64 (0.36-0ubuntu0.18.04.1), python3-acme:amd64 (0.31.0-1+ubuntu18.04.1+certbot+1), python3-configobj:amd64 (5.0.6-2+ubuntu18.04.1+certbot+1), python3-blinker:amd64 (1.4+dfsg1-0.1), ubuntu-core-launcher:amd64 (2.38+18.04), python3-lib2to3:amd64 (3.6.8-1~18.04), python3-yaml:amd64 (3.12-1build2), python3-pkg-resources:amd64 (39.0.1-2), python3-openssl:amd64 (17.5.0-1ubuntu1), dh-python:amd64 (3.20180325ubuntu2), python3-rfc3339:amd64 (1.0-4), python3-requests-toolbelt:amd64 (0.8.0-1+ubuntu18.04.1+certbot+1), python3-certbot-apache:amd64 (0.31.0-1+ubuntu18.04.1+certbot+1), sosreport:amd64 (3.6-1ubuntu0.18.04.2), python3-dbus:amd64 (1.2.6-1), python3-prettytable:amd64 (0.7.2-3), lsb-release:amd64 (9.20170808ubuntu1), python3-certbot:amd64 (0.31.0-1+ubuntu18.04.1+certbot+1), python3-gi:amd64 (3.26.1-2ubuntu1), python3-tz:amd64 (2018.3-2), unattended-upgrades:amd64 (1.1ubuntu1.18.04.11), python3-oauthlib:amd64 (2.0.6-1), apparmor:amd64 (2.12-4ubuntu5.1), command-not-found:amd64 (18.04.5), ssh-import-id:amd64 (5.7-0ubuntu1.1), python3-requests-unixsocket:amd64 (0.1.5-3), python3-jinja2:amd64 (2.10-1ubuntu0.18.04.1), python3-debian:amd64 (0.1.32), python3-gdbm:amd64 (3.6.8-1~18.04), python3-newt:amd64 (0.52.20-1ubuntu1), python3-certifi:amd64 (2018.1.18-2), cloud-guest-utils:amd64 (0.30-0ubuntu5), python3-parsedatetime:amd64 (2.4-3+ubuntu18.04.1+certbot+3), python3-chardet:amd64 (3.0.4-1), python3-problem-report:amd64 (2.20.9-0ubuntu7.6), gce-compute-image-packages:amd64 (20190124+dfsg1-0ubuntu1~18.04.0), python3-serial:amd64 (3.4-2), python3-apt:amd64 (1.6.4), python3-pyasn1:amd64 (0.4.2-3), software-properties-common:amd64 (0.96.24.32.9)
Upgrade: libpython3.6-minimal:amd64 (3.6.7-1~18.04, 3.6.8-1~18.04.1), libpython3.6-stdlib:amd64 (3.6.7-1~18.04, 3.6.8-1~18.04.1), libpython3.6:amd64 (3.6.7-1~18.04, 3.6.8-1~18.04.1), libssl-dev:amd64 (1.1.0g-2ubuntu4.3, 1.1.1-1ubuntu2.1~18.04.2), libssl1.1:amd64 (1.1.0g-2ubuntu4.3, 1.1.1-1ubuntu2.1~18.04.2)
Error: Sub-process /usr/bin/dpkg returned an error code (1)
End-Date: 2019-06-18  21:27:43

Start-Date: 2019-06-18  21:30:15
Commandline: apt install wazuh-api wazuh-manager
Requested-By: jsoref (1001)
Error: Sub-process /usr/bin/dpkg returned an error code (1)
End-Date: 2019-06-18  21:30:15

Start-Date: 2019-06-18  21:31:02
Commandline: apt remove wazuh-api wazuh-manager
Requested-By: jsoref (1001)
Remove: wazuh-api:amd64 (3.9.2-1), wazuh-manager:amd64 (3.9.2-1)
Error: Sub-process /usr/bin/dpkg returned an error code (1)
End-Date: 2019-06-18  21:31:05

Start-Date: 2019-06-18  21:31:11
Commandline: apt remove wazuh-api wazuh-manager
Requested-By: jsoref (1001)
Remove: wazuh-manager:amd64 (3.9.2-1)
Error: Sub-process /usr/bin/dpkg returned an error code (1)
End-Date: 2019-06-18  21:31:12

Start-Date: 2019-06-18  21:31:23
Commandline: apt remove wazuh-api wazuh-manager -f
Requested-By: jsoref (1001)
Remove: wazuh-manager:amd64 (3.9.2-1)
Error: Sub-process /usr/bin/dpkg returned an error code (1)
End-Date: 2019-06-18  21:31:25

Start-Date: 2019-06-18  21:32:28
Commandline: apt remove wazuh-api wazuh-manager -f
Requested-By: jsoref (1001)
Remove: wazuh-manager:amd64 (3.9.2-1)
End-Date: 2019-06-18  21:32:29

Start-Date: 2019-06-18  21:32:38
Commandline: apt install wazuh-api
Requested-By: jsoref (1001)
Install: wazuh-api:amd64 (3.9.2-1), wazuh-manager:amd64 (3.9.2-1, automatic)
Error: Sub-process /usr/bin/dpkg returned an error code (1)
End-Date: 2019-06-18  21:33:33

Start-Date: 2019-06-18  21:34:26
Commandline: apt install
Requested-By: jsoref (1001)
Error: Sub-process /usr/bin/dpkg returned an error code (1)
End-Date: 2019-06-18  21:34:26

Start-Date: 2019-06-18  21:38:03
Commandline: apt install
Requested-By: jsoref (1001)
Error: Sub-process /usr/bin/dpkg returned an error code (1)
End-Date: 2019-06-18  21:38:03

Start-Date: 2019-06-18  21:38:15
Commandline: apt install
Requested-By: jsoref (1001)
End-Date: 2019-06-18  21:38:23
```

**Comment by druizz90** [comment]
Hi @jsoref,

It seems that you have a problem with `APT` due to a broken package. You can try to solve it by following this guide:

https://itsfoss.com/dpkg-returned-an-error-code-1/

Best regards,

Demetrio.

**Comment by jsoref** [comment]
Note: I'm no longer stuck, as is clearly evident from the last log entry (which doesn't have an `Error` value).

The broken package was one of the wazuh packages, it failed to uninstall because an `rm -rf` failed because it was targeted at a mount instead of a directory.
I fixed it w/ wazuh/wazuh-packages#213

But the point is that there's some removal/reinstall path which leaves some of these items but not others, and this code blob is doing a bad job of "look before you leap" or "tolerate failed leaps" -- resulting in broken installs.

**Comment by druizz90** [comment]
Hi @jsoref,

Sorry, I thought that you was stuck. @wazuh/cicd are going to review your PR in order to merge it in our next release.

Thank you for your contribution.

Best regards,

Demetrio.
